src/components/presentational/DateRangeCoordinator/DateRangeCoordinator.tsx

```diff
@@ -12,16 +11,22 @@ export interface DateRangeCoordinatorProps {
     children?: React.ReactNode;
     innerRef?: React.Ref<HTMLDivElement>;
     sticky?: boolean;
+    useCustomNavigator?: boolean;
```

Comment:
This customization is really awesome - my only complaint about it is the usage of this boolean.  It feels redundant to have to pass this flag *in addition* to passing the actual navigator as a child.

I think it might be cleaner (or at least, maybe more misuse-proof) if this was simply `customNavigator?: React.ReactNode`.  Then line 58 below becomes `{props.customNavigator || <DateRangeNavigator... `.

Thoughts?

Reply:
Sort of an attached thought to this is that another sort of potential for mis-use here is that calling the `DateRangeContext.update` function is the single expectation of what a custom navigator must do at some point - but that fact isn't exactly *conveyed* anywhere.  It's maybe possible to convey that via types/interfaces if this isn't a boolean but a specific expected type that assures that it calls `update`?

Reply:
I like both of these suggestions.  I'll tinker with it a bit and see what I can come up with.

Reply:
Hmm.  I think we may need to leave things as-is for the highest level of customizability.

If we use the `{props.customNavigator || <DateRangeNavigator... ` approach, we force the custom navigator to be above the children.  Having it as part of the children means the navigator can be placed above, within, or below the child components.  We might be able to dynamically look for a specific type of child/descendent component in order to hide the default navigator, but having the boolean flag to disable the default navigator seems like a less fragile approach.

I also can't think of a good way to ensure the navigators call the update function on the context.  However, without calling it, they don't really do anything, so I think it would quickly become apparent to someone creating a custom navigator that they need to call something to adjust the date, which would be the update function on the context.